ompi/dpm/dpm.c

```diff
@@ -1057,15 +1057,22 @@ int ompi_dpm_dyn_finalize(void)
             comm = (ompi_communicator_t*)opal_pointer_array_get_item(&ompi_mpi_communicators,i);
             if (NULL != comm &&  OMPI_COMM_IS_DYNAMIC(comm)) {
                 objs[j++] = disconnect_init(comm);
+                // printf ("%d/%d = %p\n", j-1, ompi_comm_num_dyncomm + 1, objs[j-1]);
             }
         }
 
         if (j != ompi_comm_num_dyncomm+1) {
+            for(i=0; i<j; i++) {
```

Comment:
I don't see the point of this entire check. If j is less than ompi_comm_num_dyncomm then we should just call disconnect on what we have and continue the normal path. If however j is larger that ompi_comm_num_dyncomm we are already in big troubles as we have overwritten outside the bounds of the temporary objs array.

Reply:
Moreover there should be no opal_output(0,...) in the disconnect_waitall function.

Reply:
I will double check that tomorrow